core/chains/evm/logpoller/log_poller_internal_test.go

```diff
@@ -438,8 +443,15 @@ func TestLogPoller_Replay(t *testing.T) {
 			}()
 		})
 		ec.On("FilterLogs", mock.Anything, mock.Anything).Once().Return([]types.Log{log1}, nil).Run(func(args mock.Arguments) {
-			lp.cancel()
-			close(pass)
+			go func() {
```

Comment:
Why make this async, could it result in leakage?

Reply:
Calling `Close()` synchronously results in deadlock since it is waiting for itself to return.  I'm open to other suggestions, but this test is pretty odd. I guess I could just track it and block returning?

Reply:
From a quick read it seems like this one won't leak, but the async func above waiting on 
```
case lp.replayStart <- 4:
case <-ctx.Done():
``` 

could?

Reply:
:thinking: